packages/calypso-e2e/src/lib/blocks/block-flows/shared/block-helpers.ts

```diff
@@ -44,10 +44,29 @@ export async function validatePublishedFormFields(
  * @param {Locator} parentFormBlock Locator for the parent form block.
  * @param {FieldLabelDetails} details The details for labeling.
  */
-export async function labelFormFieldBlock( parentFormBlock: Locator, details: FieldLabelDetails ) {
-	const { blockName, accessibleLabelName, labelText } = details;
-	await parentFormBlock
+export async function labelFormFieldBlock(
+	parentFormBlock: Locator,
+	{
+		blockName,
+		accessibleLabelName,
+		labelText,
+		parentBlockName = undefined,
+		isRefactor = false,
+	}: {
+		blockName: string;
+		accessibleLabelName: string;
+		labelText: string;
+		parentBlockName?: string;
+		isRefactor?: boolean;
+	}
+) {
+	let scope = parentFormBlock;
+	if ( isRefactor && parentBlockName ) {
+		scope = scope.locator( makeSelectorFromBlockName( parentBlockName ) );
+	}
```

Comment:
See [this comment](https://github.com/Automattic/wp-calypso/pull/103974/files#r2127593717) above.